rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -243,7 +252,12 @@ where
         }
         // Report amount of deliveries stored into db
         stored_logs_metric.inc_by(stored as u64);
-        logs
+
+        if (stored as usize) < logs.len() {
+            return Err(logs);
```

Comment:
worth logging a warning if we stored insufficient logs? Probably good to include the number of deduped logs vs actually stored logs as structured log fields

Reply:
Added a warning